### PR TITLE
Align mbedtls_config.h with design

### LINKED ISSUE #9845: Align mbedtls_config.h with config-split.md
### Suggested enhancement
config-split.md redefines the mbedtls_config.h sections and config option ordering and this is not reflected in mbedtls_config.h yet.
This task is to align the content of mbedtls_config.h with config-split.md.

### Justification
Mbed TLS needs this to complete the config split work.

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description

Resolves #9845

This PR moves around the mbedtls_config entries to match the agreed design.

## PR checklist

Please remove the segment/s on either side of the | symbol as appropriate, and add any relevant link/s to the end of the line.
If the provided content is part of the present PR remove the # symbol.

- [x] **changelog** not required because: Does not change the code, only re-orders confiurations around.
- [x] **development PR** provided # | not required because: This is it.
- [x] **framework PR** not required
- [x] **3.6 PR** provided # | not required because: Only applicable to post split codebase
- [x] **2.28 PR** provided # | not required because:  Only applicable to post split codebase
- **tests**  not required because: Covered by existing testing.

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

**Review by Harry-Ramsey — APPROVED** [review]
LGTM

File: include/mbedtls/mbedtls_config.h
Comment (gabor-mezei-arm) [inline include/mbedtls/mbedtls_config.h:1]: The `repo_split.md` mentions `MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED`. Should it be removed from there?

**Review by gabor-mezei-arm — DISMISSED** [review]
Looks good to me

File: include/mbedtls/mbedtls_config.h
Comment (gilles-peskine-arm) [reply include/mbedtls/mbedtls_config.h:1]: Yes. We removed `MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED` while `config-split.md` was still under review.

That's not critical though: in any case it's clear that all the `MBEDTLS_KEY_EXCHANGE_xxx_ENABLED` will go in the same place.

**Review by Harry-Ramsey — DISMISSED** [review]
LGTM.

**Review by Harry-Ramsey — APPROVED** [review]
LGTM

**Pull request merged**